sidecar/register/healthcheck/http.go

```diff
@@ -0,0 +1,132 @@
+// Copyright 2016 IBM Corporation
+//
+//   Licensed under the Apache License, Version 2.0 (the "License");
+//   you may not use this file except in compliance with the License.
+//   You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+//   Unless required by applicable law or agreed to in writing, software
+//   distributed under the License is distributed on an "AS IS" BASIS,
+//   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//   See the License for the specific language governing permissions and
+//   limitations under the License.
+
+package healthcheck
+
+import (
+	"fmt"
+	"net/http"
+	"time"
+
+	"net/url"
+
+	"github.com/amalgam8/amalgam8/sidecar/config"
+)
+
+const (
+	defaultHTTPInterval = 30 * time.Second
+	defaultHTTPTimeout  = 5 * time.Second
+	defaultHTTPMethod   = http.MethodGet
+	defaultHTTPCode     = 200
+)
+
+// HTTP performs periodic HTTP health checks.
+type HTTP struct {
+	client *http.Client
+
+	url    string
+	method string
+	code   int
+}
+
+// NewHTTPAgent creates a new HTTP health check.
+func NewHTTPAgent(conf config.HealthCheck) (*Agent, error) {
```

Comment:
Makes sense to me. I was on the fence between doing it that way or the way I implemented it.